### PR TITLE
70016 cross staff trills

### PR DESCRIPTION
**Pull request by jimka2001**
This is discussed in issue https://musescore.org/en/node/70016#comment-540836

The core issue is that we canot use staffL = chord->staff() because that won't correspond to the noteL->line() in the case the user has pressed Shift-Cmd->Up or Shift-Cmd-Down.
Therefore we have to take staffMove() into account according to the following formula.
Staff * staffL = noteL->score()->staff(chordL->staff()->idx() + noteL->chord()->staffMove());

### CODE REVIEW

**Comment by jimka2001** [comment]
I've checked in an initial version, but it does not behave correctly unfortunately.  I have to look further.

**Comment by jimka2001** [comment]
I've fixed the problem reported above in commit 323ce72.
The problem was that when I iterate over the tracks in the staff, I have to iterate on the staff where the note is rooted, not the staff where the note appears, again a problem with staffMove()

**Comment by jimka2001** [comment]
I've also added a new test case, testTrillCrossStaff which is basically the file trill-playback-cross-staff.mscz attached to the issue https://musescore.org/en/node/70016#comment-541216

**Comment by jimka2001** [comment]
I've removed the unrelated commit, 323ce72f394f83831e79a018cc02dd7b80e09535
which will be part of a different PR.

File: libmscore/rendermidi.cpp [hunk 1]
```diff
@@ -905,7 +905,10 @@ int articulationExcursion(Note *noteL, Note *noteR, int deltastep)
       Chord *chordR = noteR->chord();
       int pitchL = noteL->pitch();
       int tickL = chordL->tick();
-      Staff * staffL = chordL->staff();
+      // we canot use staffL = chord->staff() because that won't correspond to the noteL->line()
+      //   in the case the user has pressed Shift-Cmd->Up or Shift-Cmd-Down.
+      //   Therefore we have to take staffMove() into account according to the following formula.
+      Staff * staffL = noteL->score()->staff(chordL->staff()->idx() + noteL->chord()->staffMove());
```
Comment (lasconic) [inline libmscore/rendermidi.cpp]: Check `Chord::vStaffIndex()`

**Comment by jimka2001** [comment]
Yes it seems vStaffIdx() returns exactly idx() + staffMove().  I can change that.

**Comment by lasconic** [comment]
Your git commits are not associated with an email registered at github.com. It's a pity because it makes it harder to track the actual author. I know that jimka2001 and Jim Newton are the same person but github or anyone looking into the history on github will have to figure it out. If you don't mind, could you add your epita address here https://github.com/settings/emails Thanks !

**Pull request merged**